L1Trigger/L1TGlobal/plugins/GtExternalFakeProducer.cc

```diff
@@ -0,0 +1,180 @@
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/EDGetToken.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+//#include <vector>
+#include "DataFormats/L1Trigger/interface/BXVector.h"
+
+#include "DataFormats/L1TGlobal/interface/GlobalExtBlk.h"
+
+using namespace std;
+using namespace edm;
+
+
+namespace l1t {
+
+  //
+  // class declaration
+  //
+
+  class GtExternalFakeProducer : public EDProducer {
+  public:
+    explicit GtExternalFakeProducer(const ParameterSet&);
+    ~GtExternalFakeProducer();
+
+    static void fillDescriptions(ConfigurationDescriptions& descriptions);
+
+  private:
+    virtual void produce(Event&, EventSetup const&);
+    virtual void beginJob();
+    virtual void endJob();
+    virtual void beginRun(Run const&iR, EventSetup const&iE);
+    virtual void endRun(Run const& iR, EventSetup const& iE);
+
+    // ----------member data ---------------------------
+    unsigned long long m_paramsCacheId; // Cache-ID from current parameters, to check if needs to be updated.
+    //boost::shared_ptr<const CaloParams> m_dbpars; // Database parameters for the trigger, to be updated as needed.
+    //boost::shared_ptr<const FirmwareVersion> m_fwv;
+    //boost::shared_ptr<FirmwareVersion> m_fwv; //not const during testing.
+
+    // BX parameters
+    int bxFirst_;
+    int bxLast_;
+
+    bool setBptxAND_;
+    bool setBptxPlus_;
+    bool setBptxMinus_;
+    bool setBptxOR_;
+
+  };
+
+  //
+  // constructors and destructor
+  //
+  GtExternalFakeProducer::GtExternalFakeProducer(const ParameterSet& iConfig)
+  {
+    // register what you produce
+    produces<GlobalExtBlkBxCollection>();
+
+    // Setup parameters
+    bxFirst_ = iConfig.getParameter<int>("bxFirst");
+    bxLast_  = iConfig.getParameter<int>("bxLast");
+
+    setBptxAND_ = iConfig.getParameter<bool>("setBptxAND");
+    setBptxPlus_ = iConfig.getParameter<bool>("setBptxPlus");
+    setBptxMinus_ = iConfig.getParameter<bool>("setBptxMinus");
+    setBptxOR_ = iConfig.getParameter<bool>("setBptxOR");
+
+    // set cache id to zero, will be set at first beginRun:
+    m_paramsCacheId = 0;
+  }
+
+
+  GtExternalFakeProducer::~GtExternalFakeProducer()
+  {
+  }
+
+
+
+  //
+  // member functions
+  //
+
+  // ------------ method called to produce the data ------------
+  void
+  GtExternalFakeProducer::produce(Event& iEvent, const EventSetup& iSetup)
+  {
+
+    LogDebug("GtExternalFakeProducer") << "GtExternalFakeProducer::produce function called...\n";
+
+    // Setup vectors
+    GlobalExtBlk extCond_bx;
+
+    // Set the range of BX....TO DO...move to Params or determine from param set.
+    int bxFirst = bxFirst_;
+    int bxLast  = bxLast_;
+
+
+    //outputs
+    std::auto_ptr<GlobalExtBlkBxCollection> extCond( new GlobalExtBlkBxCollection(0,bxFirst,bxLast));
+
+
+    extCond_bx.reset();
+    // Fill in some external conditions for testing
```

Comment:
I guess this is fine for testing, but shouldn't the association of each external condition to each bit come from some EventSetup object ?

(in which case, if you want to cache the mapping across events, this should be a `stream` rather than `global` producer, and it does make sense to have cache id member)